lib/bmp280.ex

```diff
@@ -120,45 +121,48 @@ defmodule BMP280 do
 
     {:ok, transport} = Transport.open(bus_name, bus_address)
 
-    state = %{
-      transport: transport,
-      calibration: nil,
-      sea_level_pa: Keyword.get(args, :sea_level_pa, @sea_level_pa),
-      sensor_type: nil
-    }
+    state =
+      %{
+        transport: transport,
+        calibration: nil,
+        sea_level_pa: Keyword.get(args, :sea_level_pa, @sea_level_pa),
+        sensor_type: nil,
+        last_measurement: nil
+      }
+      |> query_sensor()
+      |> init_sensor()
 
     {:ok, state, {:continue, :continue}}
   end
 
   @impl GenServer
   def handle_continue(:continue, state) do
-    new_state =
-      state
-      |> query_sensor()
-      |> init_sensor()
+    send(self(), :schedule_measurement)
```

Comment:
Could you change this to call a helper function to take the measurement and then start the timer? That way there will always be a measurement or an error and not the in between `nil` state.

Reply:
I think I understood the idea. I updated the code so that we measure right after initialization in series, store it to state, then schedule next.  Please let me know it something is missing.